lib/printer.js

```diff
@@ -1617,6 +1641,336 @@ function genericPrintNoParens(path, options, print) {
     case "NullTypeAnnotation":
         return fromString("null", options);
 
+    // Type Annotations for TypeScript (when using Babylon as parser)
+    case "TSType":
+        throw new Error("unprintable type: " + JSON.stringify(n.type));
+
+    case "TSNumberKeyword":
+        return fromString("number", options);
+
+    case "TSBooleanKeyword":
+        return fromString("boolean", options);
+
+    case "TSStringKeyword":
+        return fromString("string", options);
+
+    case "TSAnyKeyword":
+        return fromString("any", options);
+
+    case "TSVoidKeyword":
+        return fromString("void", options);
+
+    case "TSNullKeyword":
+        return fromString("null", options);
+
+    case "TSUndefinedKeyword":
+        return fromString("undefined", options);
+
+    case "TSNeverKeyword":
+        return fromString("never", options);
+
+    case "TSArrayType":
+        return concat([
+            path.call(print, "elementType"),
+            "[]"
+        ]);
+
+    case "TSLiteralType":
+        return path.call(print, "literal")
+
+    case "TSUnionType":
+        return fromString(" | ").join(path.map(print, "types"));
+
+    case "TSIntersectionType":
+        return fromString(" & ").join(path.map(print, "types"));
+
+    case "TSFunctionType":
+        return concat([
+            path.call(print, "typeParameters"),
+            "(",
+            fromString(", ").join(path.map(print, "parameters")),
+            ") ",
+            path.call(print, "typeAnnotation")
+        ]);
+
+    case "TSMappedType": {
+        var line = concat([
+            n.readonly ? "readonly " : "",
+            "[",
+            path.call(print, "typeParameter"),
+            "]: " ,
+            path.call(print, "typeAnnotation"),
+            ";"
+        ]);
+
+        return concat([
+            "{\n",
+            line.indent(options.tabWidth),
+            "\n}"
+        ]);
+    }
+
+    case "TSTupleType":
+        return concat([
+            "[",
+            fromString(", ").join(path.map(print, "elementTypes")),
+            "]"
+        ]);
+
+    case "TSIndexedAccessType":
+        return concat([
+            path.call(print, "objectType"),
+            "[",
+            path.call(print, "indexType"),
+            "]"
+        ]);
+
+    case "TSTypeOperator":
+        return concat([
+            path.call(print, "operator"),
+            " ",
+            path.call(print, "typeAnnotation")
+        ]);
+
+    case "TSTypeLiteral":
+        var parts = ["{"];
+        path.each(function(childPath) {
+            var lines = print(childPath);
+            parts.push(lines, ", ");
+        }, "members");
+
+        parts[parts.length - 1] = "}";
+
+        return concat(parts);
+
+    case "TSEnumMember":
+        parts = [path.call(print, "id")];
+        if (n.initializer) {
+            parts.push(
+                " = ",
+                path.call(print, "initializer")
+            );
+            parts.push();
```

Comment:
What’s this empty `.push()` doing?

Reply:
Absolutely nothing. Copy paste mistake :) Ill fix it as soon as I have some free time to get back into this later this week.